src/features/authentication-delegates/index.ts

```diff
@@ -0,0 +1,118 @@
+import * as jwt from 'jsonwebtoken';
+
+import { sbvrUtils, permissions } from '@balena/pinejs';
+import { findUser } from '../../infra/auth/auth';
+import { createSessionToken } from '../../infra/auth/jwt';
+import { Application, RequestHandler } from 'express';
+import { SetupOptions } from '../..';
+import { captureException } from '../../infra/error-handling';
+const { api } = sbvrUtils;
+const { rootRead } = permissions;
+
+// our API model
+type AuthenticationDelegateModel = {
+    uuid: string,
+    public_key: string,
+}
+
+// our JSON token format
+type AuthenticationDelegateModelResults = { [key: string]: any } & Partial<AuthenticationDelegateModel>;
+
+export type DelegateToken = {
+    [key: string]: any,
+    delegateUuid?: string,
+    userId?: string,
+} | string | null;
+
+export async function extractTrustedUserId(token: string) {
+    const decodedToken: DelegateToken = jwt.decode(token);
```

Comment:
1) decode the token without validating it. This gets the delegate UUID is provided in the JWT claims.

Reply:
@richbayliss wouldn't you need the secret to decode a jwt?

Reply:
never mind, found my answer.

Reply:
> never mind, found my answer.

cool, but yeah, the secret is an asymmetric key type, so we decode the JWT to JSON (and then an object in TS) to determine the UUID of the delegate to decide which public key to use to validate the JWT signature :+1: